pkg/database/database.go

```diff
@@ -901,7 +901,7 @@ func (d *db) History(req *schema.HistoryRequest) (*schema.Entries, error) {
 		}
 
 		md, val, err := d.st.ReadValue(tx, key)
-		if err != nil {
+		if err != nil && err != store.ErrKeyNotFound {
```

Comment:
I think we need something like that:
```suggestion
		if error.Is(err, store.ErrKeyNotFound) {
			list.Entries[i] = &schema.Entry{
				Tx:       txID,
				Key:      req.Key,
				Metadata: schema.KVMetadataToProto(.... something indicating an expired record ...),
				Value:    []byte{},
			}
			continue
		}
		if err != nil {
```

or at least protection against using nil in `TrimPrefix` below

Reply:
as discussed offline, a dedicated expiration error was defined, making it possible to retrieve actual metadata